### DISCUSSION TITLE
Profile conformance

### DISCUSSION THREAD

**Discussion by BenjaminKraus4711** [FreeOpcUa/python-opcua#1331] [General]
Does any one know which profile is implemented by the FreeOpcUa server?

**Comment by AndreasHeine** [comment]
tbh currently no one!

**Reply by BenjaminKraus4711** [reply]
Ok, thx! I had already thought that it is so. Any idea what is necessary to fullfill at least the Nano Embedded profiles criteria? 

Another question: I saw that the "Server" node is quite empty, especially ServerCapabilities, ServerDiagnostics and ServerRedundancy. Is there any way to fill those nodes from by the Server class or do I need to implement those things on my own?

**Reply by swamper123** [reply]
There is a list from the foundation here:

https://profiles.opcfoundation.org/v104/Reporting/index.htm?ModifyProfile.aspx?ProfileID=53d63aa6-9c4a-4b6e-b825-a75363a2515b

What especially is missing is the View Service, even in [our better maintained opcua-asyncio](https://github.com/FreeOpcUa/opcua-asyncio).

**Reply by oroulet** [reply]
Fylling these things is mainly some administrative work. PR welcome

**Reply by swamper123** [reply]
Administrative work with a positive advertisment effect. ;)

**Reply by oroulet** [reply]
yes it would be nice if someone would look at it. ...
